Library/Homebrew/utils/ruby.sh

```diff
@@ -1,8 +1,15 @@
-test-ruby () {
-  [[ ! -x $1 ]] && { echo "false"; return 1; }
-  "$1" --enable-frozen-string-literal --disable=gems,did_you_mean,rubyopt -rrubygems -e \
+test_ruby () {
+  if [[ ! -x $1 ]]
+  then
+    return 1
+  fi
+
+  local ruby_status
+  ruby_status=$("$1" --enable-frozen-string-literal --disable=gems,did_you_mean,rubyopt -rrubygems -e \
     "puts Gem::Version.new(RUBY_VERSION.to_s.dup).to_s.split('.').first(2) == \
-          Gem::Version.new('$required_ruby_version').to_s.split('.').first(2)" 2>/dev/null
+          Gem::Version.new('$required_ruby_version').to_s.split('.').first(2)" 2>/dev/null)
+
+  test "$ruby_status" = true
```

Comment:
Does this mean the `== true` comparisons below are now unnecessary?

Reply:
right. the idea was to make it look like so

```sh
if test_ruby "$ruby_exec"
then
    ...
```

If we make `test_ruby` output text to stdout, it'll have to be

```sh
if test_ruby "$ruby_exec" &>/dev/null
then
    ...
```

Reply:
Alternatively you could do the larger refactor so the `stdout` isn't necessary.

Reply:
> Alternatively you could do the larger refactor so the `stdout` isn't necessary.

I'm OK with that if @sjackman is OK with keeping `test-ruby` in Homebrew/install in-sync with the function here.

Reply:
Yep!

Reply:
Marking as resolved because there is an agreement on the next steps (making stdout unnecessary).